File: data/src/main/java/com/marvelcomics/brito/data/datasource/remote/response/ComicResponse.java
package com.marvelcomics.brito.data.datasource.remote.response;

import com.google.gson.annotations.SerializedName;

public class ComicResponse {

    @SerializedName("id")
    private int id;
    @SerializedName("title")
    private String title;
    @SerializedName("description")
    private String description;
    @SerializedName("thumbnail")
    private ThumbnailResponse thumbnailResponse;

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    public String getDescription() {
        return description;
    }

    public void setDescription(String description) {
        this.description = description;
    }

    public ThumbnailResponse getThumbnailResponse() {
        return thumbnailResponse;
    }

    public void setThumbnailResponse(ThumbnailResponse thumbnailResponse) {
        this.thumbnailResponse = thumbnailResponse;
    }
}

File: data-remote/src/main/java/com/marvelcomics/brito/data_remote/datasource/response/ComicResponse.kt
package com.marvelcomics.brito.data_remote.datasource.response

import com.google.gson.annotations.SerializedName
import com.marvelcomics.brito.data_remote.datasource.response.model.RemoteMarvelContainerResponse
import com.marvelcomics.brito.entity.ComicEntity

class ComicResponse {
    @SerializedName("id")
    var id = 0

    @SerializedName("title")
    var title: String? = null

    @SerializedName("description")
    var description: String? = null

    @SerializedName("thumbnail")
    var thumbnailResponse: ThumbnailResponse? = null
}

fun RemoteMarvelContainerResponse<ComicResponse>.fromResponseToEntity(): List<ComicEntity>? {
    return this.remoteMarvelDataResponse?.results?.map { comicResponse ->
        ComicEntity(
            comicResponse.id,
            comicResponse.title,
            comicResponse.description,
            comicResponse.thumbnailResponse?.fromResponseToEntity()
        )
    }
}
